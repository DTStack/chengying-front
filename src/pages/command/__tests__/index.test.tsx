import * as React from 'react';
import Command from '../index';
import { render, RenderResult, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

const defaultProps = {
  history: {},
  location: {},
  match: {},
};

/**
 * 测试header组件的reducer
 */
describe('command unit test', () => {
  let wrapper: RenderResult;

  beforeEach(() => {
    wrapper = render(<Command {...defaultProps} />);
  });

  afterEach(cleanup);

  test('command page render search', () => {
    const searchBox = wrapper.getByPlaceholderText('请输入搜索关键词');
    expect(searchBox).toBeInTheDocument();
  });

  test('command list render', () => {
    const columns = wrapper.container.getElementsByClassName(
      'ant-table-column-title'
    );
    // columns render
    for (const i in columns) {
      const titles = ['', '对象', '操作', '状态', '开始时间', '持续时间'];
      const column = columns[i];
      if (column instanceof HTMLElement) {
        expect(column.textContent).toBe(titles[i]);
      }
    }
  });

  // rerender
  test('command snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
