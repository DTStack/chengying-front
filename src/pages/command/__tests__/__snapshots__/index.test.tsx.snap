// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command unit test command snapshot 1`] = `
<DocumentFragment>
  <div
    class="scriptecho-page"
  >
    <div
      class="echo-form clearfix mb-12"
    >
      <span
        class="mr-20"
      >
        对象：
        <div
          class="ant-select-show-search ant-select-auto-complete ant-select ant-select-combobox ant-select-enabled ant-select-allow-clear"
        >
          <div
            aria-autocomplete="list"
            aria-controls="test-uuid"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-select-selection
            ant-select-selection--single"
            role="combobox"
          >
            <div
              class="ant-select-selection__rendered"
            >
              <ul>
                <li
                  class="ant-select-search ant-select-search--inline"
                >
                  <div
                    class="ant-select-search__field__wrap"
                  >
                    <span
                      class="ant-select-search__field ant-input-affix-wrapper"
                    >
                      <input
                        class="ant-input"
                        placeholder="请输入搜索关键词"
                        type="text"
                        value=""
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        <i
                          aria-label="icon: search"
                          class="anticon anticon-search"
                        >
                          <svg
                            aria-hidden="true"
                            class=""
                            data-icon="search"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                            />
                          </svg>
                        </i>
                      </span>
                    </span>
                    <span
                      class="ant-select-search__field__mirror"
                    >
                       
                    </span>
                  </div>
                </li>
              </ul>
            </div>
            <span
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <i
                aria-label="icon: down"
                class="anticon anticon-down ant-select-arrow-icon"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </i>
            </span>
          </div>
        </div>
      </span>
      <span
        class="mr-20"
      >
        操作：
        <div
          class="ant-select ant-select-enabled ant-select-allow-clear"
        >
          <div
            aria-autocomplete="list"
            aria-controls="test-uuid"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-select-selection
            ant-select-selection--single"
            role="combobox"
            tabindex="0"
          >
            <div
              class="ant-select-selection__rendered"
            >
              <div
                class="ant-select-selection__placeholder"
                style="display: block; user-select: none;"
                unselectable="on"
              >
                请选择命令
              </div>
              <div
                class="ant-select-search ant-select-search--inline"
                style="display: none;"
              >
                <div
                  class="ant-select-search__field__wrap"
                >
                  <input
                    autocomplete="off"
                    class="ant-select-search__field"
                    value=""
                  />
                  <span
                    class="ant-select-search__field__mirror"
                  >
                     
                  </span>
                </div>
              </div>
            </div>
            <span
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <i
                aria-label="icon: down"
                class="anticon anticon-down ant-select-arrow-icon"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </i>
            </span>
          </div>
        </div>
      </span>
      <span
        class="mr-20"
      >
        状态：
        <div
          class="ant-select ant-select-enabled ant-select-allow-clear"
        >
          <div
            aria-autocomplete="list"
            aria-controls="test-uuid"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-select-selection
            ant-select-selection--single"
            role="combobox"
            tabindex="0"
          >
            <div
              class="ant-select-selection__rendered"
            >
              <div
                class="ant-select-selection__placeholder"
                style="display: block; user-select: none;"
                unselectable="on"
              >
                请选择状态
              </div>
              <div
                class="ant-select-search ant-select-search--inline"
                style="display: none;"
              >
                <div
                  class="ant-select-search__field__wrap"
                >
                  <input
                    autocomplete="off"
                    class="ant-select-search__field"
                    value=""
                  />
                  <span
                    class="ant-select-search__field__mirror"
                  >
                     
                  </span>
                </div>
              </div>
            </div>
            <span
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <i
                aria-label="icon: down"
                class="anticon anticon-down ant-select-arrow-icon"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </i>
            </span>
          </div>
        </div>
      </span>
      <span
        class="mr-20"
      >
        开始时间：
        <span
          class="ant-calendar-picker"
          style="width: 350px;"
          tabindex="0"
        >
          <span
            class="ant-calendar-picker-input ant-input"
          >
            <input
              class="ant-calendar-range-picker-input"
              placeholder="Start date"
              readonly=""
              tabindex="-1"
              value=""
            />
            <span
              class="ant-calendar-range-picker-separator"
            >
               ~ 
            </span>
            <input
              class="ant-calendar-range-picker-input"
              placeholder="End date"
              readonly=""
              tabindex="-1"
              value=""
            />
            <i
              aria-label="icon: calendar"
              class="anticon anticon-calendar ant-calendar-picker-icon"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="calendar"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                />
              </svg>
            </i>
          </span>
        </span>
      </span>
    </div>
    <div
      class="ant-table-wrapper dt-table-fixed-base"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div>
          <div
            class="ant-spin ant-spin-spinning ant-table-without-pagination ant-table-spin-holder"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
          </div>
        </div>
        <div
          class="ant-spin-container ant-spin-blur"
        >
          <div
            class="ant-table ant-table-default ant-table-empty ant-table-fixed-header ant-table-scroll-position-left ant-table-layout-fixed"
          >
            <div
              class="ant-table-content"
            >
              <div
                class="ant-table-scroll"
              >
                <div
                  class="ant-table-header"
                >
                  <table
                    class=""
                  >
                    <colgroup>
                      <col
                        style="width: 30px; min-width: 30px;"
                      />
                      <col
                        style="width: 20%; min-width: 20%;"
                      />
                      <col />
                      <col />
                      <col />
                      <col
                        style="width: 12%; min-width: 12%;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-row-cell-break-word"
                        >
                          <span
                            class="ant-table-header-column"
                          >
                            <div>
                              <span
                                class="ant-table-column-title"
                              />
                              <span
                                class="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          class="ant-table-row-cell-break-word"
                        >
                          <span
                            class="ant-table-header-column"
                          >
                            <div>
                              <span
                                class="ant-table-column-title"
                              >
                                对象
                              </span>
                              <span
                                class="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          class=""
                        >
                          <span
                            class="ant-table-header-column"
                          >
                            <div>
                              <span
                                class="ant-table-column-title"
                              >
                                操作
                              </span>
                              <span
                                class="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          class=""
                        >
                          <span
                            class="ant-table-header-column"
                          >
                            <div>
                              <span
                                class="ant-table-column-title"
                              >
                                状态
                              </span>
                              <span
                                class="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          class=""
                        >
                          <span
                            class="ant-table-header-column"
                          >
                            <div>
                              <span
                                class="ant-table-column-title"
                              >
                                开始时间
                              </span>
                              <span
                                class="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                        <th
                          class="ant-table-row-cell-break-word ant-table-row-cell-last"
                        >
                          <span
                            class="ant-table-header-column"
                          >
                            <div>
                              <span
                                class="ant-table-column-title"
                              >
                                持续时间
                              </span>
                              <span
                                class="ant-table-column-sorter"
                              />
                            </div>
                          </span>
                        </th>
                      </tr>
                    </thead>
                  </table>
                </div>
                <div
                  class="ant-table-body"
                  style="overflow-y: scroll;"
                  tabindex="-1"
                >
                  <table
                    class=""
                  >
                    <colgroup>
                      <col
                        style="width: 30px; min-width: 30px;"
                      />
                      <col
                        style="width: 20%; min-width: 20%;"
                      />
                      <col />
                      <col />
                      <col />
                      <col
                        style="width: 12%; min-width: 12%;"
                      />
                    </colgroup>
                    <tbody
                      class="ant-table-tbody"
                    />
                  </table>
                </div>
                <div
                  class="ant-table-placeholder"
                >
                  <div
                    class="ant-empty ant-empty-normal"
                  >
                    <div
                      class="ant-empty-image"
                    >
                      <svg
                        height="41"
                        viewBox="0 0 64 41"
                        width="64"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                          transform="translate(0 1)"
                        >
                          <ellipse
                            cx="32"
                            cy="33"
                            fill="#F5F5F5"
                            rx="32"
                            ry="7"
                          />
                          <g
                            fill-rule="nonzero"
                            stroke="#D9D9D9"
                          >
                            <path
                              d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                            />
                            <path
                              d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                              fill="#FAFAFA"
                            />
                          </g>
                        </g>
                      </svg>
                    </div>
                    <p
                      class="ant-empty-description"
                    >
                      No Data
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
